src/features/addTweetActions.ts

```diff
@@ -194,36 +235,129 @@ export const maybeAddTweetActions = makeBTDModule(({settings, TD, jq}) => {
     navigator.clipboard.writeText(mediaUrls);
   });
 
+  modifyMustacheTemplate(TD, 'stream_item.mustache', (string) =>
+    string.replace(
+      'data-tweet-id="{{#getMainTweet}}{{id}}{{/getMainTweet}}"',
+      'data-tweet-id="{{#getMainTweet}}{{id}}{{/getMainTweet}}" data-tweet-user-id="{{#getMainUser}}{{id}}{{/getMainUser}}{{^getMainUser}}{{targetTweet.user.id}}{{/getMainUser}}"'
+    )
+  );
+
+  TD.mustaches['text/follow_action.mustache'] =
+    '{{^following}} {{_i}}Follow @{{screenName}}{{/i}} {{/following}} {{#following}} {{_i}}Unfollow @{{screenName}}{{/i}} {{/following}}';
+
+  modifyMustacheTemplate(TD, 'status/tweet_single.mustache', (string) =>
+    string.replace(
+      '{{#isFavorite}}is-favorite{{/isFavorite}}',
+      '{{#isFavorite}}is-favorite{{/isFavorite}} {{#getMainUser}}{{#following}}following{{/following}}{{/getMainUser}}'
+    )
+  );
+
+  modifyMustacheTemplate(TD, 'status/tweet_detail.mustache', (string) =>
+    string
+      .replace(
+        'data-tweet-id="{{#getMainTweet}}{{id}}{{/getMainTweet}}"',
+        'data-tweet-id="{{#getMainTweet}}{{id}}{{/getMainTweet}}" data-tweet-user-id="{{getMainUser.id}}"'
+      )
+      .replace(
+        '{{#isFavorite}}is-favorite{{/isFavorite}}',
+        '{{#isFavorite}}is-favorite{{/isFavorite}} {{#getMainUser}}{{#following}}following{{/following}}{{/getMainUser}}'
+      )
+  );
+
+  // Similar to `TD.services.TwitterStatus.prototype.setFavorite`
+  TD.services.TwitterUser.prototype.setFollowing = function (following: boolean) {
+    this.following = following;
+    const addedClass = following ? 'icon-following' : 'icon-follow';
+    const removedClass = following ? 'icon-follow' : 'icon-following';
+    jq(`[data-tweet-user-id="${this.id}"][data-account-key="${this.account.getKey()}"]`)
```

Comment:
Not a blocker, more of me thinking out loud: every chirp "seen" by BTD [gets an uuid](https://github.com/eramdam/BetterTweetDeck/blob/e4634531dc78878a659033e632e76a02f6b9407a/src/services/chirpHandler.ts#L90) set on them, I feel like there needs to be a way to use that in that case as well 😅

Reply:
Do you mean it is better to access every chirp node whose `.following` class is toggled using UUID?
I cannot think of a way to do that because we need to toggle the class of every node whose author is followed by the BTD user at once, not only of one node...

Reply:
Yeah, there's currently no way to get the UUID of a given chirp without already having its DOM node. I need to figure out a way. But also you're right that in this case it wouldn't be useful because we may need to update more than one node 😅